### PR TITLE
Add a badge of code quality

### PR DESCRIPTION
**Pull request by asdf2014**
## Description

Add a badge of code quality to README:

[Language grade: Java](https://lgtm.com/projects/g/apache/iotdb/context:java)

<hr>

This PR has:
- [x] been self-reviewed.
    - [x] concurrent read
    - [x] concurrent write
    - [x] concurrent read and write 
- [x] added documentation for new or modified features or behaviors.
- [x] added Javadocs for most classes and all non-trivial methods. 
- [x] added or updated version, __license__, or notice information
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [x] added integration tests.
- [x] been tested in a test IoTDB cluster.

<hr>

### CODE REVIEW

**Comment by jixuan1989** [comment]
Hi, pls update jdk from 8 to 11 on both code coverage and sonar in `.github/workflows/main-linux.yml` :

image

see: https://sonarcloud.io/documentation/appendices/end-of-support/

**Comment by asdf2014** [comment]
Hi, @jixuan1989 . Thanks for your comment. It's done.

File: .github/workflows/main-linux.yml [hunk 1]
```diff
@@ -57,7 +57,7 @@ jobs:
           -Pcode-coverage
       - name: SonarCloud Report
         #jdk8 is faster to run UT/IT
-        if: ${{ success() &&  matrix.java == '8'}}
+        if: ${{ success() &&  matrix.java == '11'}}
```
Comment (HTHou) [inline .github/workflows/main-linux.yml:60]: emmm...how about remove or modify this comment?

**Review by HTHou — APPROVED** [review]
LGTM

File: .github/workflows/main-linux.yml (hunk 1 above)
Comment (asdf2014) [reply .github/workflows/main-linux.yml]: Hi, @HTHou . I agree with you. It's done.

**Pull request merged**